### ISSUE TITLE
[bitnami/redis-cluster] scaleup doesn't re-distribute slots for masters

### ISSUE THREAD

**Issue by tpolekhin** [bitnami/charts#3853]
**Which chart**:
redis-cluster-3.2.4

**Describe the bug**
After scaling up Redis cluster slots aren't re-distributed across new number of masters

**To Reproduce**
1. Deploy redis-cluster with 3 masters and 0 slaves
2. Make sure cluster is deployed and slots distributed across 3 masters
3. Upgrade chart with 6 masters and 0 slaves
4. New masters joined the cluster but have no slots assigned

**Expected behavior**
Whenever we scale redis-cluster slots should be re-distributed across new number of masters

**Version of Helm and Kubernetes**:

- Output of `helm version`:

```
version.BuildInfo{Version:"v3.3.3", GitCommit:"55e3ca022e40fe200fbc855938995f40b2a68ce0", GitTreeState:"dirty", GoVersion:"go1.15.2"}
```

- Output of `kubectl version`:

```
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.6", GitCommit:"dff82dc0de47299ab66c83c626e08b245ab19037", GitTreeState:"clean", BuildDate:"2020-07-15T23:30:39Z", GoVersion:"go1.14.4", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.12+IKS", GitCommit:"d09005b98837bb6061c0f643a27383c02b003205", GitTreeState:"clean", BuildDate:"2020-09-16T21:47:16Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}
```

**Additional context**
initial deployment
```
>>> Performing Cluster Check (using node 172.21.190.221:6379)
M: b76d429758e309e0acbf9cb5900116ef7c5951cd 172.21.190.221:6379
   slots:[0-5460] (5461 slots) master
M: ad5847b0859f7878ed6495c27b969c89019dddd7 172.21.218.162:6379
   slots:[10923-16383] (5461 slots) master
M: 6da7a064e51fecf19241457239b8cdf58ab50dd8 172.21.69.21:6379
   slots:[5461-10922] (5462 slots) master
[OK] All nodes agree about slots configuration.
>>> Check for open slots...
>>> Check slots coverage...
[OK] All 16384 slots covered.
Cluster correctly created
```

after scaleup
```
>>> Performing Cluster Check (using node 172.21.190.223:6379)
M: b76d429758e309e0acbf9cb5900116ef7c5951cd 172.21.190.223:6379
   slots:[0-5460] (5461 slots) master
M: 6da7a064e51fecf19241457239b8cdf58ab50dd8 172.21.69.23:6379
   slots:[5461-10922] (5462 slots) master
M: 191072af67981b63a91921cf321d5c52c8179ca2 172.21.204.151:6379
   slots: (0 slots) master
M: b5313545b459d0cdf2d719e632a04e0ed5ec1d5f 172.21.166.28:6379
   slots: (0 slots) master
M: ad5847b0859f7878ed6495c27b969c89019dddd7 172.21.233.211:6379
   slots:[10923-16383] (5461 slots) master
[OK] All nodes agree about slots configuration.
>>> Check for open slots...
>>> Check slots coverage...
[OK] All 16384 slots covered.
```

**Comment by dani8art** [comment]
Hi @tpolekhin we are experiencing some issue when trying to update replicas using this chart, please see https://github.com/bitnami/charts/issues/3743

We have opened an internal task to do further investigations, we will keep you updated.

**Comment by alemorcuq** [comment]
Fixed in #8619
